core/src/stack/outputs.rs

```diff
@@ -84,6 +88,27 @@ impl StackOutputs {
     // PUBLIC ACCESSORS
     // --------------------------------------------------------------------------------------------
 
+    /// Returns the value located at the specified position on the stack
+    pub fn get_stack_item(&self, idx: usize) -> Option<Felt> {
+        self.stack.get(idx).map(|&felt| felt.into())
+    }
```

Comment:
I wonder if this function should be fallible. If we try and access an element with `idx > self.stack.len() - 1` then we should return an error?

Reply:
I prefer to return an `Option` when there's only one error possible (here: out of bounds). If we returned `OutputError` for example, and a caller wanted to match on the error, then they would need to handle error cases that could never happen. Then, if we create a new error type with one variant, I consider that "type bloat" because it doesn't convey anymore information than an `Option` would, but introduces a new name. 

I believe this is a convention widely used in the standard library, such as with [`slice::get`](https://doc.rust-lang.org/std/primitive.slice.html#method.get).

Reply:
I updated the comments to document what `None` means here